ports/espressif/common-hal/espulp/ULP.c

```diff
@@ -34,7 +36,14 @@ void espulp_reset(void) {
     ulp_used = false;
 }
 
-void common_hal_espulp_ulp_run(espulp_ulp_obj_t *self, uint32_t *program, size_t length, uint32_t pin_mask) {
+void common_hal_espulp_ulp_set_wakeup_period(espulp_ulp_obj_t *self, size_t period_index, uint32_t period_us) {
+    int _errno = ulp_set_wakeup_period(period_index, period_us);
+    if (_errno != ESP_OK) {
+        mp_raise_ValueError(MP_ERROR_TEXT("Invalid parameters"));
```

Comment:
Please don't add a new error string if [something close already exists](https://github.com/adafruit/circuitpython/blob/c47f013b9f7596fdaec67a71a891c96ec1f9df84/locale/circuitpython.pot#L1210)

```suggestion
        mp_raise_ValueError(MP_ERROR_TEXT("Invalid argument"));
```

There is also "Invalid %q" that you can use if you know which of the two is invalid. That'd be more specific.

Reply:
Thanks for the feedback! I made some changes to hopefully improve the error messages. Here, I opted for `CHECK_ESP_RESULT` which would raise an `Invalid Argument` error if the `period_index` is wrong. I could have used `Invalid %q` to specify that it's the `period_index` that is wrong. But that would have meant a new string, and here the user can easily figure out which argument is wrong based on the description of the function.

Reply:
This looks good. `Invalid %q` is the same string regardless of what `MP_QSTR` you give it. So you could also do that in a follow up PR.